usr/local/www/diag_logs_filter.php

```diff
@@ -144,282 +175,308 @@ function getGETPOSTsettingvalue($settingname, $default)
-				<?php endif; ?>
-				<?php echo htmlspecialchars($filterent['interface']);?></td>
-			  <?php
-			  if ($config['syslog']['filterdescriptions'] === "1")
-				echo("<td class=\"listMRr nowrap\">".find_rule_by_number_buffer($filterent['rulenum'],$filterent['tracker'],$filterent['act'])."</td>");
-
-			  $int = strtolower($filterent['interface']);
-			  $proto = strtolower($filterent['proto']);
-			  if($filterent['version'] == '6') {
-				$ipproto = "inet6";
-				$filterent['srcip'] = "[{$filterent['srcip']}]";
-				$filterent['dstip'] = "[{$filterent['dstip']}]";
-			  } else {
-					$ipproto = "inet";
-			  }
-
-			  $srcstr = $filterent['srcip'] . get_port_with_service($filterent['srcport'], $proto);
-			  $src_htmlclass = str_replace(array('.', ':'), '-', $filterent['srcip']);
-			  $dststr = $filterent['dstip'] . get_port_with_service($filterent['dstport'], $proto);
-			  $dst_htmlclass = str_replace(array('.', ':'), '-', $filterent['dstip']);
-			  ?>
-			  <td class="listMRr nowrap">
-				<img onclick="javascript:resolve_with_ajax('<?php echo "{$filterent['srcip']}"; ?>');" title="<?=gettext("Click to resolve");?>" class="ICON-<?= $src_htmlclass; ?>" border="0" src="/themes/<?= $g['theme']; ?>/images/icons/icon_log.gif" alt="Icon Reverse Resolve with DNS"/>
-				<a href="easyrule.php?<?php echo "action=block&amp;int={$int}&amp;src={$filterent['srcip']}&amp;ipproto={$ipproto}"; ?>" title="<?=gettext("Easy Rule: Add to Block List");?>" onclick="return confirm('<?=gettext("Do you really want to add this BLOCK rule?")?>')">
-				<img border="0" src="/themes/<?= $g['theme']; ?>/images/icons/icon_block_add.gif" alt="Icon Easy Rule: Add to Block List" /></a>
-				<?php echo $srcstr . '<span class="RESOLVE-' . $src_htmlclass . '"></span>';?>
-			  </td>
-			  <td class="listMRr nowrap">
-				<img onclick="javascript:resolve_with_ajax('<?php echo "{$filterent['dstip']}"; ?>');" title="<?=gettext("Click to resolve");?>" class="ICON-<?= $dst_htmlclass; ?>" border="0" src="/themes/<?= $g['theme']; ?>/images/icons/icon_log.gif" alt="Icon Reverse Resolve with DNS"/>
-				<a href="easyrule.php?<?php echo "action=pass&amp;int={$int}&amp;proto={$proto}&amp;src={$filterent['srcip']}&amp;dst={$filterent['dstip']}&amp;dstport={$filterent['dstport']}&amp;ipproto={$ipproto}"; ?>" title="<?=gettext("Easy Rule: Pass this traffic");?>" onclick="return confirm('<?=gettext("Do you really want to add this PASS rule?")?>')">
-				<img border="0" src="/themes/<?= $g['theme']; ?>/images/icons/icon_pass_add.gif" alt="Icon Easy Rule: Pass this traffic" /></a>
-				<?php echo $dststr . '<span class="RESOLVE-' . $dst_htmlclass . '"></span>';?>
-			  </td>
-			  <?php
-				if ($filterent['proto'] == "TCP")
-					$filterent['proto'] .= ":{$filterent['tcpflags']}";
+<?php
+	if ($config['syslog']['filterdescriptions'])
+		buffer_rules_load();
+
+	$rowIndex = 0;
+	foreach ($filterlog as $filterent) {
+		$evenRowClass = $rowIndex % 2 ? " listMReven" : " listMRodd";
```

Comment:
thankfully, this variable isn't used anymore, please remove it. You can also replace these lines with simply:

```php
foreach ($filterlog as $rowindex => $filterent) {
```

It'll properly set rowindex, no need to define and update it manually